spec/models/repository/git_spec.rb

```diff
@@ -0,0 +1,363 @@
+                  '7234cb2750b63f47bff735edc50a1c0a433c2518'
+                ])
+
+          changesets = instance.latest_changesets('images', '899a15dba', 1)
+          expect(changesets.map(&:revision))
+            .to eq(['899a15dba03a3b350b89c3f537e4bbe02a03cdc9'])
+
+          changesets = instance.latest_changesets('README', '899a15dba')
+          expect(changesets.map(&:revision))
+            .to eq([
+                  '899a15dba03a3b350b89c3f537e4bbe02a03cdc9',
+                  '7234cb2750b63f47bff735edc50a1c0a433c2518'
+                  ])
+
+          changesets = instance.latest_changesets('README', '899a15dba', 1)
+          expect(changesets.map(&:revision))
+            .to eq(['899a15dba03a3b350b89c3f537e4bbe02a03cdc9'])
+        end
+
+        it 'should fetch changesets with tag' do
+          changesets = instance.latest_changesets('images', 'tag01.annotated')
+          expect(changesets.map(&:revision))
+            .to eq([
+                  '899a15dba03a3b350b89c3f537e4bbe02a03cdc9',
+                  '7234cb2750b63f47bff735edc50a1c0a433c2518'
+                  ])
+
+          changesets = instance.latest_changesets('README', '899a15dba', 1)
+          expect(changesets.map(&:revision))
+            .to eq(['899a15dba03a3b350b89c3f537e4bbe02a03cdc9'])
+
+          changesets = instance.latest_changesets('README', 'tag01.annotated')
+          expect(changesets.map(&:revision))
+            .to eq([
+                  '899a15dba03a3b350b89c3f537e4bbe02a03cdc9',
+                  '7234cb2750b63f47bff735edc50a1c0a433c2518'
+                  ])
+
+          changesets = instance.latest_changesets('README', 'tag01.annotated', 1)
+          expect(changesets.map(&:revision))
+            .to eq(['899a15dba03a3b350b89c3f537e4bbe02a03cdc9'])
+        end
+
+        it 'should fetch changesets with path, branch, and limit' do
+          changesets = instance.latest_changesets('images', 'test_branch')
+          expect(changesets.map(&:revision))
+            .to eq([
+                  '899a15dba03a3b350b89c3f537e4bbe02a03cdc9',
+                  '7234cb2750b63f47bff735edc50a1c0a433c2518'
+                  ])
+
+          changesets = instance.latest_changesets('images', 'test_branch', 1)
+          expect(changesets.map(&:revision))
+            .to eq(['899a15dba03a3b350b89c3f537e4bbe02a03cdc9'])
+
+          changesets = instance.latest_changesets('README', 'test_branch')
+          expect(changesets.map(&:revision))
+            .to eq([
+                  '713f4944648826f558cf548222f813dabe7cbb04',
+                  '61b685fbe55ab05b5ac68402d5720c1a6ac973d1',
+                  '899a15dba03a3b350b89c3f537e4bbe02a03cdc9',
+                  '7234cb2750b63f47bff735edc50a1c0a433c2518'
+                  ])
+
+          changesets = instance.latest_changesets('README', 'test_branch', 2)
+          expect(changesets.map(&:revision))
+            .to eq([
+                  '713f4944648826f558cf548222f813dabe7cbb04',
```

Comment:
Use 2 spaces for indentation in an array, relative to the start of the line where the left bracket is.